Client/Application/AppDelegate.swift

```diff
@@ -364,6 +358,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         case .NewPrivateTab:
             browserViewController.applyPrivateModeTheme(force: true)
             browserViewController.openBlankNewTabAndFocus(isPrivate: true)
+        case .OpenLastBookmark:
+            // open bookmark in a non-private browsing tab
+            browserViewController.applyNormalModeTheme(force: true)
+            // find out if bookmarked URL is currently open
+            // if so, open to that tab,
+            // otherwise, create a new tab with the bookmarked URL
+            if let urlString = userData?["bookmarkURL"] as? String,
+                let urlToOpen = NSURL(string: urlString) {
```

Comment:
Nit: We can condense this a bit by using our asURL extension method:

`if let urlToOpen = userData?["bookmarkURL"]?.asURL`